src/main/scala/nl.knaw.dans.easy.bagstore/component/BagStoreComponent.scala

```diff
@@ -184,13 +180,16 @@ trait BagStoreComponent {
                   Files.copy(path, outputStream)
                 })
             }
-            else if (allEntries.isEmpty) Failure(NoSuchItemException(itemId))
             else Failure(NoRegularFileException(itemId))
           }
         } yield ()
       }
     }
 
+    private def createDirSpecs(bagDir: BaseDir, itemPath: BaseDir, fileIds: Seq[FileId]): Try[Seq[EntrySpec]] = Try {
+      fileIds.filter(_.isDirectory).map(createEntrySpec(None, bagDir, itemPath, _))
```

Comment:
use a `.collect` instead of `.filter(...).map(...)`

Reply:
I see you already did that in your PR.